src/main/kotlin/com/github/catppuccin/jetbrains_icons/providers/JavaIconProvider.kt

```diff
@@ -16,46 +16,90 @@ import com.intellij.ui.RowIcon
+   * @param flags Additional flags for icon retrieval.
+   * @return The icon for the element, or null if no suitable icon is found.
+   */
   override fun getIcon(@NotNull element: PsiElement, @IconFlags flags: Int): Icon? {
-    if (!PluginSettingsState.instance.javaSupport) return icons.java
-
-    if (element !is PsiClass) return null
-
-    val virtualFile = PsiUtilCore.getVirtualFile(element) ?: return null
-    if (!virtualFile.name.endsWith(".java")) return null
+    return when {
+      !PluginSettingsState.instance.javaSupport -> icons.java
+      element !is PsiClass -> null
+      PsiUtilCore.getVirtualFile(element)?.name?.endsWith(".java") != true -> null
+      else -> getJavaClassIcon(element)
+    }
+  }
 
+  /**
+   * Gets the appropriate icon for a Java class, including static and visibility markers.
+   *
+   * @param element The PsiClass to get an icon for.
+   * @return The icon for the Java class.
+   */
+  private fun getJavaClassIcon(element: PsiClass): Icon {
     val baseIcon = getJavaIcon(element)
-    val staticMark = getStaticMark(element)
+    val iconWithStaticMark = addStaticMarkIfNeeded(element, baseIcon)
+    return addVisibilityIconIfNeeded(element, iconWithStaticMark)
+  }
 
-    val icon =
-      when {
-        staticMark != null -> {
-          LayeredIcon(2).apply {
-            setIcon(baseIcon, 0)
-            setIcon(staticMark, 1)
-          }
-        }
-        else -> baseIcon
+  /**
+   * Adds a static mark to the icon if the class is static.
+   *
+   * @param element The PsiClass to check for static modifier.
+   * @param baseIcon The base icon to add the static mark to.
+   * @return The icon with static mark added if needed.
+   */
+  private fun addStaticMarkIfNeeded(element: PsiClass, baseIcon: Icon): Icon {
+    val staticMark = getStaticMark(element)
+    return if (staticMark != null) {
+      LayeredIcon(2).apply {
+        setIcon(baseIcon, 0)
+        setIcon(staticMark, 1)
       }
+    } else {
+      baseIcon
+    }
+  }
 
+  /**
+   * Adds a visibility icon to the class icon if visibility icons are enabled in the project view.
+   *
+   * @param element The PsiClass to get the visibility for.
+   * @param icon The icon to add the visibility icon to.
+   * @return The icon with visibility icon added if needed.
+   */
+  private fun addVisibilityIconIfNeeded(element: PsiClass, icon: Icon): Icon {
     val visibilityIconsEnabled =
       ProjectView.getInstance(element.project)?.isShowVisibilityIcons("ProjectPane") == true
-
-    return when {
-      visibilityIconsEnabled -> {
-        RowIcon(2).apply {
-          setIcon(icon, 0)
-          getVisibilityIcon(element)?.let { setIcon(it, 1) }
-        }
+    return if (visibilityIconsEnabled) {
+      RowIcon(2).apply {
+        setIcon(icon, 0)
+        getVisibilityIcon(element)?.let { setIcon(it, 1) }
       }
-      else -> icon
+    } else {
+      icon
     }
   }
 
+  /**
+   * Gets the static mark icon if the class is static.
```

Comment:
I can tell it gets a static mark icon from the name `getStaticMark`, what is a static mark?